### PR TITLE
Fix for machine_cpu_cores being removed from cadvisor metrics

### PR DESCRIPTION
**Pull request by dnoland1**
See also https://github.com/rancher/rancher/issues/29292

### CODE REVIEW

File: pkg/controllers/managementuser/alert/initdata.go
Comment (aiyengar2) [inline pkg/controllers/managementuser/alert/initdata.go:287]: This is a deprecated metric that comes from `kube-state-metrics`. It will officially be removed in v2.x based on https://github.com/kubernetes/kube-state-metrics/issues/974.

However, this should be okay since:
- The change only impacts Alerting V1, not Alerting V2. Alerting V2 does not use MetricRules afaik (@codyrancher we aren't using this for https://github.com/rancher/dashboard/issues/812 right?)
- Alerting V2 is still on version 1.9.x of kube-state-metrics and is deprecated, so that is unlikely to change.

@guangbochen I'm adding you as a reviewer to also get a sign-off since this is a Monitoring / Alerting V1 issue. I want to confirm that using this metric is okay and that we have no plans of upgrading the `kube-state-metrics` version of Monitoring / Alerting V1.

**Review by aiyengar2 — APPROVED** [review]
@dnoland1 LGTM but we also need a backport to 2.4 since this will primarily only affect users who are in 2.4 (as users in 2.5 are expected to use Alerting V2, which doesn't use MetricRules at all).

Also, we should wait on a sign-off from the China team about whether this metric is okay to be used.

**Comment by dnoland1** [comment]
@aiyengar2 Since this is only an issue on k8s 1.19+ and v2.4.x only supports up to k8s 1.18, is it really necessary to backport?

**Comment by aiyengar2** [comment]
> @aiyengar2 Since this is only an issue on k8s 1.19+ and v2.4.x only supports up to k8s 1.18, is it really necessary to backport?

Hmm so this brings into question whether this fix is necessary at all. If we were planning to backport to 2.4, it would make sense to forward port it to 2.5. However, if it's only going to impact users who are using Monitoring / Alerting V1 in Rancher 2.5 (which is considered deprecated in favor of Monitoring / Alerting V2), then should we push out a fix at all or just encourage users to switch to V2 for k8s 1.19+ clusters?

@maggieliu are we still actively supporting Monitoring / Alerting V1 in Rancher 2.5?

**Comment by maggieliu** [comment]
@dnoland1 @aiyengar2 Don't need to backport to 2.4 because k8s 1.19 is not supported in 2.4. As for whether or not we need this fix at all: technically we are not actively support v1 anymore because it's deprecated. However, if there is a high priority request from customer, then we will still put in fixes.

**Comment by aiyengar2** [comment]
> @dnoland1 @aiyengar2 Don't need to backport to 2.4 because k8s 1.19 is not supported in 2.4. As for whether or not we need this fix at all: technically we are not actively supporting v1 anymore because it's deprecated. However, if there is a high priority request from customer, then we will still put in fixes.

@dnoland1 is this a high priority fix? Otherwise, I think we should avoid changing the Monitoring V1 for just Rancher 2.5, especially since this is a soon-to-be-removed metric from kube-state-metrics.

**Comment by dnoland1** [comment]
It's nice to have. The CPU load out of box v1 alert will not work on k8s 1.19. Users can manually workaround the issue by replacing the metric in their alert.

File: pkg/controllers/managementuser/alert/initdata.go
Comment (guangbochen) [reply pkg/controllers/managementuser/alert/initdata.go:287]: this is LGTM since for v1 monitoring the latest v1 monitoring chart is using v1.9.7 `kube-state-metrics` and the metric is supported.
https://github.com/rancher/system-charts/blob/dev-v2.5/charts/rancher-monitoring/v0.2.0/values.yaml#L187

**Review by guangbochen — APPROVED** [review]

**Review by aiwantaozi — DISMISSED** [review]

File: pkg/controllers/managementuser/alert/initdata.go
Comment (aiyengar2) [reply pkg/controllers/managementuser/alert/initdata.go:287]: @dnoland1, based on https://github.com/kubernetes/kubernetes/issues/95204#issuecomment-719445180, should the correct, non-deprecated call be the following?
```suggestion
						Expression:     `sum(node_load1) by (node)  / sum(kube_node_status_capacity{resource="cpu"}) by (node) * 100`,
```

**Review by aiyengar2 — CHANGES_REQUESTED** [review]
Unsure if this change should be merged, waiting for confirmation on whether Monitoring V1 should be supported in k8s 1.19 clusters (which is what this fix is for)

**Review by aiyengar2 — APPROVED** [review]
LGTM, seems like we still want to support V1 tools in Rancher 2.5 so this should be good to merge as long as it has been tested + we get a committer review.

@dnoland1 can you make sure that the newly changed expression `kube_node_status_capacity{resource="cpu"}` outputs the same thing you expected from the deprecated metric? Not sure if you already tested it again before pushing out the commit I suggested.

**Comment by dnoland1** [comment]
I tried putting the expression directly into the UI and worked as expected.

**Comment by aiyengar2** [comment]
@dnoland1 similar to https://github.com/rancher/rancher/pull/30339, we need this PR to be opened from `dnoland1:somebranch` to `master` to avoid running into build failures because of Drone. Can you open another PR from your fork and close out this PR?

**Comment by dnoland1** [comment]
Done - https://github.com/rancher/rancher/pull/30343

**Pull request closed without merging**
